ypathgen/pathgen.go

```diff
@@ -718,14 +743,19 @@ func generateDirectorySnippet(directory *ygen.Directory, directories map[string]
 	var methodBuf strings.Builder
 
 	// Output struct snippets.
-	structData := getStructData(directory, pathStructSuffix, generateWildcardPaths)
+	structData := getStructData(directory, pathStructSuffix, schemaStructPkgAccessor, generateWildcardPaths)
 	if ygen.IsFakeRoot(directory.Entry) {
 		// Fakeroot has its unique output.
 		if err := goPathFakeRootTemplate.Execute(&structBuf, structData); err != nil {
 			return GoPathStructCodeSnippet{}, util.AppendErr(errs, err)
 		}
-	} else if err := goPathStructTemplate.Execute(&structBuf, structData); err != nil {
-		return GoPathStructCodeSnippet{}, util.AppendErr(errs, err)
+	} else {
+		if err := goPathStructTemplate.Execute(&structBuf, structData); err != nil {
+			return GoPathStructCodeSnippet{}, util.AppendErr(errs, err)
+		}
+		if err := goPathAndStructHelperTemplate.Execute(&structBuf, structData); err != nil {
```

Comment:
Should generating the path and struct helper be optional? In general we've tried to have these other features that are new types of output flag-controlled in the library - even if we have it as on by default, it'd probably be nice to have something whereby a user can turn this off if they don't want it to save on generated code size.